File: Leetcode/0001-1000/0201-0300/0242. Valid Anagram - Word Count.cpp
// https://leetcode.com/problems/valid-anagram/


class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.size() != t.size()) {
            return false;
        }
        int cnts[26] = {0};
        for (int i = 0; i < s.size(); i++) {
            cnts[s[i] - 'a']++;
        }
        for (int i = 0; i < t.size(); i++) {
            cnts[t[i] - 'a']--;
        }
        for (int i = 0; i < 26; i++) {
            if (cnts[i] != 0) {
                return false;
            }
        }
        return true;
    }
};

File: Leetcode/0001-1000/0201-0300/0242. Valid Anagram - Word Count.py
# https://leetcode.com/problems/valid-anagram/


class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False

        cnts = [0] * 26
        for c in s:
            cnts[ord(c) - 97] += 1
        for c in t:
            cnts[ord(c) - 97] -= 1
        return all(map(lambda x: x == 0, cnts))
